### PR TITLE
DPL: adding DataAllocator method to check validity of a certain output

### PR DESCRIPTION
**Pull request by matthiasrichter**
`DataAllocator::isAllowed` can be used to check for conditional outputs.

### CODE REVIEW

File: Framework/Core/src/DataAllocator.cxx [hunk 1]
```diff
@@ -251,5 +251,15 @@ Output DataAllocator::getOutputByBind(OutputRef&& ref)
   O2_BUILTIN_UNREACHABLE();
 }
 
+bool DataAllocator::isValid(OutputSpec const& query)
```
Comment (ktf) [inline Framework/Core/src/DataAllocator.cxx]: How about `isAllowed`?

Comment (ktf) [reply Framework/Core/src/DataAllocator.cxx]: Also, shouldn't this use "Output" rather than `OutputSpec`?

Comment (matthiasrichter) [reply Framework/Core/src/DataAllocator.cxx]: `isAllowed` is as good as `isValid`, can change this. For the query type I was also thinking about `Output` but then I wanted to have this close to what one adds to the output specs, and I wanted to use `ConcreteDataTypeMatcher`. Probably not a good idea, as one has to specify `Output` with all three coordinates anyhow in the API methods. So why not doing this consistently in the check. You are right.

Comment (matthiasrichter) [reply Framework/Core/src/DataAllocator.cxx]: the code is adapted acordingly

**Comment by matthiasrichter** [comment]
GPU CI seems to be broken, unrelated
```
-- Configuring done
CMake Error in GPU/GPUTracking/Base/cuda/CMakeLists.txt:
  Target "O2lib-GPUTrackingCUDA" requires the language dialect "CUDA17" (with
  compiler extensions), but CMake does not know the compile flags to use to
  enable it.


CMake Error in Detectors/ITSMFT/ITS/tracking/cuda/CMakeLists.txt:
  Target "O2lib-ITStrackingCUDA" requires the language dialect "CUDA17" (with
  compiler extensions), but CMake does not know the compile flags to use to
  enable it.
```

File: Framework/Core/src/DataAllocator.cxx (hunk 1 above)
Comment (ktf) [reply Framework/Core/src/DataAllocator.cxx]: Depends what the purpose of the API is. If it is to check if a given message is a valid / allowed one, then ConcreteDataTypeMatcher / Output is the correct thing to have. If the goal is to check whether a given rule is among the ones present, then OutputSpec is what to use. Regarding the name, I guess my comment was more about "isValid" not referring to the DataAllocator itself, but to the passed object. `allowsFor` maybe it's a better name for what I meant.

**Pull request merged**